scripts/ingest-tarball.sh

```diff
@@ -73,7 +77,36 @@ then
     error "the contents type in the filename (${contents_type_dir}) does not match the contents type in the tarball (${tar_contents_type_dir})."
 fi
 
-# Ingest the tarball to the repository, use "versions" as base dir for the ingestion
+# If this is a compat layer tarball, we need to check if it's an update, and if so, use a special procedure
+if [ "${tar_contents_type_dir}" = "compat" ]
+then
+    tar_first_file="$(tar tf "${tar_file}" | head -n 1)"
+    # Get OS and architecture from path, which should look like: <version>/compat/<os>/<arch>
+    compat_os=$(echo "${tar_first_file}" | cut -d / -f 3)
+    compat_arch=$(echo "${tar_first_file}" | cut -d / -f 4)
```

Comment:
Maybe we need to be a bit more careful here, and have a hard check for `compat_os` being `linux` and `compat_arch` being one of `x86_64`, `aarch64` or `ppc64le` (maybe also add `riscv64`), to avoid nasty surprises?

Reply:
Done by adding a function that checks if these are set and have valid values.